src/server/game/Spells/Spell.cpp

```diff
@@ -5664,10 +5664,11 @@ SpellCastResult Spell::CheckCast(bool strict, uint32* param1 /*= nullptr*/, uint
                 if (res != SPELL_CAST_OK)
                     return res;
 
-                // chance for fail at lockpicking attempt
-                // second check prevent fail at rechecks
-                if (skillId != SKILL_NONE && (!m_selfContainer || ((*m_selfContainer) != this)))
+                // some tradeskills can fail - but herbalism, mining and skinning cannot as of patch 3.1.0
+                if (skillId != SKILL_NONE && skillId != SKILL_HERBALISM && skillId != SKILL_MINING && skillId != SKILL_SKINNING && (!m_selfContainer || ((*m_selfContainer) != this)))
```

Comment:
Don't check SKILL_SKINNING here - it uses SPELL_EFFECT_SKINNING, not SPELL_EFFECT_OPEN_LOCK